kani-compiler/src/kani_middle/transform/check_uninit/mod.rs

```diff
@@ -0,0 +1,423 @@
+                            user_ty: None,
+                            const_: MirConst::from_bool(value),
+                        }),
+                    ],
+                    ret_place,
+                );
+            }
+            PointeeLayout::Slice { element_layout } => {
+                // Since `str`` is a separate type, need to differentiate between [T] and str.
+                let (slicee_ty, diagnostic) = match pointee_info.ty().kind() {
+                    TyKind::RigidTy(RigidTy::Slice(slicee_ty)) => {
+                        (slicee_ty, "KaniSetSlicePtrInitialized")
+                    }
+                    TyKind::RigidTy(RigidTy::Str) => {
+                        (Ty::unsigned_ty(UintTy::U8), "KaniSetStrPtrInitialized")
+                    }
+                    _ => unreachable!(),
+                };
+                let set_ptr_initialized_instance = resolve_mem_init_fn(
+                    get_mem_init_fn_def(tcx, diagnostic, &mut self.mem_init_fn_cache),
+                    element_layout.len(),
+                    slicee_ty,
+                );
+                let layout_operand =
+                    mk_layout_operand(body, source, operation.position(), &element_layout);
+                try_mark_new_bb_as_skipped(&operation, body, skip_first);
+                body.add_call(
+                    &set_ptr_initialized_instance,
+                    source,
+                    operation.position(),
+                    vec![
+                        ptr_operand,
+                        layout_operand,
+                        Operand::Constant(ConstOperand {
+                            span: source.span(body.blocks()),
+                            user_ty: None,
+                            const_: MirConst::from_bool(value),
+                        }),
+                    ],
+                    ret_place,
+                );
+            }
+            PointeeLayout::TraitObject => {
+                try_mark_new_bb_as_skipped(&operation, body, skip_first);
+                let reason = "Kani does not support reasoning about memory initialization of pointers to trait objects.";
+                self.unsupported_check(tcx, body, source, operation.position(), reason);
+            }
+        };
+    }
+
+    fn unsupported_check(
+        &self,
+        tcx: TyCtxt,
+        body: &mut MutableBody,
+        source: &mut SourceInstruction,
+        position: InsertPosition,
+        reason: &str,
+    ) {
+        let span = source.span(body.blocks());
+        let rvalue = Rvalue::Use(Operand::Constant(ConstOperand {
+            const_: MirConst::from_bool(false),
+            span,
+            user_ty: None,
+        }));
+        let result = body.new_assignment(rvalue, source, position);
+        body.add_check(tcx, &self.check_type, source, position, result, reason);
+    }
+}
+
+/// Generate a bit array denoting padding vs data bytes for a layout.
```

Comment:
Can you be a bit more specific? Maybe an example.

I'm assuming this is generating an array where `false` means padding and `true` means data?